doc/source/tools.rst

```diff
@@ -1,30 +1,259 @@
+This is essentially a list of instructions that tells
+Docker how to create an image. It might tell Docker which base layers
+you want to include in an image, as well as some extra dependencies that
+you need in the image. Think of it like a recipe that tells Docker how
+to create an image.
+
+Containers
+**********
+
+You can "run" a container image, and it creates a container for you.
+A container is a particular instantiation of a container image. This means
+that it actually exists on a computer. It is a self-contained
+computational environment that is constructed according to the layers
+that are inside of the Container Image. However, because it is now
+running on the computer, it can do other useful things like talk to other
+Docker containers or communicate via the internet. 
+
+
+Kubernetes
+----------
+
+`Kubernetes <https://kubernetes.io/>`_ is a service that runs on cloud infrastructures. It provides a single point of contact with the machinery
+of your cluster deployment, and allows a user to specify the computational requirements that they need (e.g., how many machines, how many CPUs
+per machine, how much RAM).
+Then, it handles the resources on the cluster and ensures that
+these resources are always available. If something goes down, kubernetes
+will try to automatically bring it back up.
+ 
+Kubernetes can only manage the computing resources that it is
+given. This means that it generally can **not** create new resources on its
+own (with the exception of disk space).
+
+There are three main types of objects in Kubernetes:
+
+Processes
+*********
+Are any program that is running on a machine. For example,
+a Jupyter Notebook creates several processes that handle the
+execution of code and the display in the browser.
+
+Pods
+****
+
+Pods are essentially a collection of one or more *containers* that
+run together. You can think of them as a way of combining containers
+that, as a group, accomplish some goal.
+
+For example, say you want to create a web server that is open to the
+world, but you also want authentication so that only a select group
+of users can access it. You could use a single pod with two containers.
+
+* One that does the authentication. It would have something like Apache
+  specified in its container image, and would be connected to the
+  outside world.
+* One that receives information from the authentication container, and
+  does something fancy with it (maybe it runs a python process).
+
+This is useful because it lets you compatmentalize the components of the
+service that you want to run, which makes things easier to manage and
+keeps things more stabl.
+
+Deployments
+***********
+
+A deployment is a collection of pods on kubernetes. It is how kubernetes
+knows exactly what containers and what machines need to be running at all
+times. For example, if you have two pods: one that does the authenticating
+described above, and another that manages a database, you can specify both
```

Comment:
what's a good use case then to explain the value of a deployment? I'm just trying to motivate the idea that deployments aren't always 1 pod per deployment